src/main/php/com/github/mustache/MustacheEngine.class.php

```diff
@@ -90,25 +95,24 @@ public function withHelpers(array $helpers) {
   }
 
   /**
-   * Compile a template.
+   * Compile a template
    *
-   * @param  string $template The template, as a string
+   * @param  string|com.github.mustache.templates.Source $source The template source
    * @param  string $start Initial start tag, defaults to "{{"
    * @param  string $end Initial end tag, defaults to "}}"
    * @param  string $indent Indenting level, defaults to no indenting
    * @return com.github.mustache.Template
    */
-  public function compile($template, $start= '{{', $end= '}}', $indent= '') {
-    return new Template('<string>', $this->parser->parse(
-      new StringTokenizer($template),
-      $start,
-      $end,
-      $indent
-    ));
+  public function compile($source, $start= '{{', $end= '}}', $indent= '') {
+    if ($source instanceof Source) {
+      return $source->compile($this->parser, $start, $end, $indent);
+    } else {
+      return new Template('<string>', $this->parser->parse(new StringTokenizer($source), $start, $end, $indent));
```

Comment:
This could also be:

```php
$source= new Tokens('<string>', new StringTokenizer($source));
return $source->compile($this->parser, $start, $end, $indent); 
```

...but the inlined version saves us a couple of nanoseconds here:)